src/store/mod.rs

```diff
@@ -57,7 +57,18 @@ impl Store {
         Ok(result)
     }
 
-    pub fn insert_account(&self, account: &Account) -> Result<(), StoreError> {
+    pub fn insert_account(&mut self, account: &Account) -> Result<(), StoreError> {
```

Comment:
I think `insert_account_with_metadata()` for this function and `insert_account()` instead of `insert_account_initial()` might be a little more descriptive names.